### PR TITLE
UI: Location fetch on load

### PR DESCRIPTION
**Pull request by chelshaw**
- Updates the frontend to use ember-data for location query and find by coord
- Updates the backend to wrap location responses in common Response dataclass
- All in all, this allows the URLs to be shared between friends without an error on the frontend

### CODE REVIEW

File: backend/src/fun_forecast_backend/main.py [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 app.add_middleware(
     CORSMiddleware,
-    allow_origins=["http://myfunforecast.com", "http://beta.myfunforecast.com", "https://beta.myfunforecast.com"],
```
Comment (chelshaw) [inline backend/src/fun_forecast_backend/main.py:26]: For allowing the local frontend to connect to local backend

File: backend/src/fun_forecast_backend/shared/dataclasses.py [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import datetime
 from dataclasses import dataclass
 from enum import Enum
-from typing import Optional
+from typing import Optional, Any
 
 
+@dataclass
+class ResponseObject:
```
Comment (chelshaw) [inline backend/src/fun_forecast_backend/shared/dataclasses.py:8]: The frontend framework expects the ApiResponse to have a shape of:
```
{
data?: Any
meta?: Any
errors?: Any
}
```
But I wasn't sure how to allow optional keys like that.

Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/shared/dataclasses.py:8]: Good question - I think we can make them nullable and you can check if the value is present

Comment (chelshaw) [reply backend/src/fun_forecast_backend/shared/dataclasses.py:8]: How does one make a key nullable? I did a cursory google and didn't see a simple answer

Comment (shayaanjagtap) [reply backend/src/fun_forecast_backend/shared/dataclasses.py:8]: We should make an ABC response object and inherit it to specific responses imo. I can work on that

**Review by shayaanjagtap — APPROVED** [review]

**Pull request merged**